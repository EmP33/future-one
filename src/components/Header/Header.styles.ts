import styled from "styled-components";

export const Wrapper = styled.header`
  width: 100%;
  display: grid;
  grid-template-columns: max-content 1fr;
  min-height: 100px;
  margin-top: 2rem;

  @media only screen and (max-width: 1200px) {
    min-height: 0;
    margin-top: 1rem;
  }

  .header-logo {
    align-self: end;
    margin-right: 1.25rem;

    @media only screen and (max-width: 1300px) {
      margin-right: 1rem;
    }

    & span {
      font-size: 34px;
      font-weight: bold;
    }
  }

  & nav {
    width: 100%;
    display: flex;
    justify-content: space-between;
    align-self: start;

    @media only screen and (max-width: 1200px) {
      display: none;
    }

    & ul {
      display: flex;
      column-gap: 2.5rem;
      align-items: center;

      @media only screen and (max-width: 1300px) {
        column-gap: 2rem;
      }

      & a {
        color: var(--color-gray);
        text-decoration: none;

        &.active {
          color: var(--color-text);
          text-decoration: underline;
        }

        &:hover {
          text-decoration: underline;
        }
      }
    }
  }

  & button {
    font-size: 2rem;
    justify-self: end;
    align-self: center;
    display: flex;
    align-items: center;
    cursor: pointer;
    border: none;
    background: transparent;

    @media only screen and (min-width: 1200px) {
      display: none;
    }
  }
`;
