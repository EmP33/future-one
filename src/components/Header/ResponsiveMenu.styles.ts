import styled from "styled-components";

export const Wrapper = styled.nav<{ showMenu: boolean }>`
  width: 100vw;
  height: 100vh;
  position: fixed;
  background-color: var(--color-green);
  top: 0;
  left: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  transition: all 0.2s linear;
  transform: ${({ showMenu }) =>
    showMenu ? "translateX(0)" : "translateX(100%)"};

  @media only screen and (min-width: 1200px) {
    display: none;
  }

  & button {
    font-size: 2rem;
    position: absolute;
    cursor: pointer;
    top: 4%;
    right: 5.5%;
    border: none;
    background: transparent;

    @media only screen and (max-width: 900px) {
      top: 4%;
      right: 4%;
    }
  }

  & ul {
    display: flex;
    flex-direction: column;

    & a {
      display: block;
      color: var(--color-text);
      text-decoration: none;
      text-transform: uppercase;
      position: relative;
      padding: 1rem 3rem;
      font-weight: bold;

      &:hover::after {
        width: 100%;
      }

      &::after {
        content: "";
        position: absolute;
        left: 0;
        bottom: 0;
        width: 0;
        height: 1px;
        background-color: var(--color-text);
        transition: all 0.2s linear;
      }
    }
  }
`;
