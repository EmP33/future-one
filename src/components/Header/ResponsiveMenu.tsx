import React from "react";
import { Wrapper } from "./ResponsiveMenu.styles";
import { Link } from "react-router-dom";
import { IoMdClose } from "react-icons/io";

interface IProps {
  showMenu: boolean;
  toggleShowMenu: () => void;
}

const ResponsiveMenu: React.FC<IProps> = ({ showMenu, toggleShowMenu }) => {
  return (
    <Wrapper showMenu={showMenu}>
      <button onClick={toggleShowMenu}>
        <IoMdClose role="button" />
      </button>
      <ul>
        <li>
          <Link to="/" className="active">
            Overview
          </Link>
        </li>
        <li>
          <Link to="/">About</Link>
        </li>
        <li>
          <Link to="/">Resources</Link>
        </li>
        <li>
          <Link to="/">Network</Link>
        </li>
        <li>
          <Link to="/">Cloud</Link>
        </li>
        <li>
          <Link to="/">Download</Link>
        </li>
        <li>
          <Link to="/">Contact</Link>
        </li>
      </ul>
    </Wrapper>
  );
};

export default ResponsiveMenu;
