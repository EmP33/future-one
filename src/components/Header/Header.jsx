import React, { useState } from "react";
import ReactDOM from "react-dom";
import { Wrapper } from "./Header.styles";
import { Link } from "react-router-dom";
import { PrimaryButton } from "../../Global.styles";
// Icons
import { FiMenu } from "react-icons/fi";
import ResponsiveMenu from "./ResponsiveMenu";

const Header = () => {
  const [showMenu, setShowMenu] = useState(false);

  const toggleShowMenu = () => {
    setShowMenu((prev) => !prev);
  };

  return (
    <>
      <Wrapper>
        <div className="header-logo" data-aos="fade-down">
          <span>Future one</span>
        </div>
        <nav>
          <ul>
            <li data-aos="fade-down" data-aos-delay={100}>
              <Link to="/" className="active">
                Overview
              </Link>
            </li>
            <li data-aos="fade-down" data-aos-delay={200}>
              <Link to="/">About</Link>
            </li>
            <li data-aos="fade-down" data-aos-delay={300}>
              <Link to="/">Resources</Link>
            </li>
            <li data-aos="fade-down" data-aos-delay={400}>
              <Link to="/">Network</Link>
            </li>
            <li data-aos="fade-down" data-aos-delay={500}>
              <Link to="/">Cloud</Link>
            </li>
            <li data-aos="fade-down" data-aos-delay={600}>
              <Link to="/">Download</Link>
            </li>
            <li data-aos="fade-down" data-aos-delay={700}>
              <Link to="/">Contact</Link>
            </li>
          </ul>
          <PrimaryButton to="/" data-aos="fade-left" data-aos-delay={1000}>
            Contact Us
          </PrimaryButton>
        </nav>
        <button onClick={toggleShowMenu}>
          <FiMenu />
        </button>
      </Wrapper>
      {ReactDOM.createPortal(
        <ResponsiveMenu showMenu={showMenu} toggleShowMenu={toggleShowMenu} />,
        document.getElementById("modal-root")
      )}
    </>
  );
};

export default Header;
