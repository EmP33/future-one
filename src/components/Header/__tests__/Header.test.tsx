import Header from "../Header";
import ShallowRenderer from "react-test-renderer/shallow";

beforeEach(() => {
  // setup a DOM element as a render target
  let container = document.createElement("div");
  container.setAttribute("id", "modal-root");
  document.body.appendChild(container);
});

describe("<Header />", () => {
  it("should render properly", () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Header />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
