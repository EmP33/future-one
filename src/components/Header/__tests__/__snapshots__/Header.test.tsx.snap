// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render properly 1`] = `
<React.Fragment>
  <styled.header>
    <div
      className="header-logo"
      data-aos="fade-down"
    >
      <span>
        Future one
      </span>
    </div>
    <nav>
      <ul>
        <li
          data-aos="fade-down"
          data-aos-delay={100}
        >
          <Link
            className="active"
            to="/"
          >
            Overview
          </Link>
        </li>
        <li
          data-aos="fade-down"
          data-aos-delay={200}
        >
          <Link
            to="/"
          >
            About
          </Link>
        </li>
        <li
          data-aos="fade-down"
          data-aos-delay={300}
        >
          <Link
            to="/"
          >
            Resources
          </Link>
        </li>
        <li
          data-aos="fade-down"
          data-aos-delay={400}
        >
          <Link
            to="/"
          >
            Network
          </Link>
        </li>
        <li
          data-aos="fade-down"
          data-aos-delay={500}
        >
          <Link
            to="/"
          >
            Cloud
          </Link>
        </li>
        <li
          data-aos="fade-down"
          data-aos-delay={600}
        >
          <Link
            to="/"
          >
            Download
          </Link>
        </li>
        <li
          data-aos="fade-down"
          data-aos-delay={700}
        >
          <Link
            to="/"
          >
            Contact
          </Link>
        </li>
      </ul>
      <Styled(Link)
        data-aos="fade-left"
        data-aos-delay={1000}
        to="/"
      >
        Contact Us
      </Styled(Link)>
    </nav>
    <button
      onClick={[Function]}
    >
      <FiMenu />
    </button>
  </styled.header>
  Object {
    "$$typeof": Symbol(react.portal),
    "children": <ResponsiveMenu
      showMenu={false}
      toggleShowMenu={[Function]}
    />,
    "containerInfo": <div
      id="modal-root"
    />,
    "implementation": null,
    "key": null,
  }
</React.Fragment>
`;
